code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Drugs.dm

```diff
@@ -0,0 +1,242 @@
+		M.dizziness = max(150, M.dizziness)
+		M.make_dizzy(10)
+		M.druggy = max(M.druggy, 40)
+		if(prob(15))
+			M.emote(pick("twitch", "giggle"))
+
+/datum/reagent/raskara_dust
+	name = "Raskara Dust"
+	id = "raskara_dust"
+	description = "A powdery narcotic found in the gang-ridden slums of Biesel and Sol. Known for it's relaxing poperties that cause trance like states when inhaled. Casual users tend to snort or inhale, while hardcore users inject."
+	reagent_state = SOLID
+	color = "#AABBAA"
+	overdose = REAGENTS_OVERDOSE
+	taste_description = "baking soda"
+	metabolism = REM * 0.1
+	breathe_met = REM * 0.2
+	ingest_met = REM * 0.3
+
+/datum/reagent/raskara_dust/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
+	M.add_chemical_effect(CE_PAINKILLER, 10)
+	M.drowsyness += 1 * removed
+
+/datum/reagent/raskara_dust/affect_breathe(var/mob/living/carbon/M, var/alien, var/removed)
+	M.add_chemical_effect(CE_PAINKILLER, 25)
+	M.drowsyness += 2 * removed
+	if(prob(5))
+		M.emote("cough")
+
+/datum/reagent/raskara_dust/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
+	M.add_chemical_effect(CE_PAINKILLER, 50)
+	M.drowsyness += 3 * removed
+	if(prob(5))
+		M.emote("twitch")
+
+/datum/reagent/night_juice
+	name = "Nightlife"
+	id = "night_juice"
+	description = "A liquid narcotic commonly used by the more wealthy drug-abusing citizens of the Eridani Federation. Works as a potent stimulant that causes extreme awakefulness. Lethal in high doses."
+	reagent_state = LIQUID
+	color = "#FFFF44"
+	taste_description = "salted sugar"
+	metabolism = REM * 0.2
+	breathe_met = REM * 0.1
+	breathe_mul = 0.5
+	ingest_mul = 0.125
+	var/special_counter = 0
+
+/datum/reagent/night_juice/affect_blood(var/mob/living/carbon/human/M, var/alien, var/removed)
+	if(!istype(M))
+		return
+
+	special_counter += (volume/10)*removed
+
+	M.make_jittery(5 + special_counter)
+	M.drowsyness = max(0,M.drowsyness - (1 + special_counter*0.1))
+	if(special_counter > 5)
+		M.add_chemical_effect(CE_SPEEDBOOST, 1)
+		M.apply_effect(1 + special_counter*0.25, STUTTER)
+		M.druggy = max(M.druggy, special_counter*0.25)
+		M.hallucination = max(M.hallucination, special_counter*5 - 100)
+		M.make_jittery(special_counter)
+		if(prob(special_counter))
+			M.emote("twitch")
+		var/obj/item/organ/H = M.internal_organs_by_name["heart"]
+		H.take_damage(special_counter * removed * 0.025)
+
+
+/datum/reagent/guwan_painkillers
+	name = "Tremble"
+	id = "guwan_painkillers"
+	description = "An ancient tribal Unathi narcotic based on the outer gel layer of the seeds of poisonous flower. The chemical itself acts as a very potent omni-healer when consumed, however as the chemical metabolizes, it causes immense and cripling pain."
+	reagent_state = LIQUID
+	color = "#FFFF44"
+	taste_description = "sunflower seeds"
+	metabolism = REM * 0.25
+	overdose = 10
+
+/datum/reagent/guwan_painkillers/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
+	affect_ingest(M,alien,removed*0.5)
```

Comment:
I'm pretty sure after a chemical is ingested, it moves into the bloodstream? This is probably not a good idea.

Reply:
That's not how it works. I overridden affect_ingest so that this doesn't happen.